### DISCUSSION TITLE
is trade.entry_price wrong? Open/Close prices doesn't match entry price

### DISCUSSION THREAD

**Discussion by ghost** [kernc/backtesting.py#466] [Q&A] [answered]
Hello, 

I am having some troubles with the reported entry price. I am using the EURUSD data set. 

def next(self):

     if self.trades:
        print(f"Trades {self.trades[0].entry_price} {self.trades[0].entry_time}")
     if criteria_to_buy:
         self.buy()
         print(f"Orders {self.orders} {self.data.Open[-1]} {self.data.Close[-1]})
     elif criteria_to_sell:
        self.position.close() 

I cannot match the entry_price value with any of the near values(Open or Close) around the reported entry_time row in the dataset. 

What could be wrong? 

By the way I really like the project, Thank you very much!!

**Comment by ghost** [comment, marked as answer]
Never mind I just figured it out. I have to add the commission to the next open price, after this the price_entry matches.

Thanks!

**Reply by kernc** [reply]
Indeed, the oddity you observe is due to entry price adjustment for set comission:
https://github.com/kernc/backtesting.py/blob/7671f044d554249b608b3241dd7cca37ec5a35d1/backtesting/backtesting.py#L745-L750

**Reply by ghost** [reply]
Thank you very much!
